### PR TITLE
[Android] Fix the `run_debug_test_android` device lab test

### PR DESCRIPTION
**Pull request by loic-sharma**
Fix the `run_debug_test_android` device lab test. I missed running this locally before merging https://github.com/flutter/flutter/pull/116798 🤦

This was tested by ensuring the following passed locally:

```
cd dev/devicelab
../../bin/cache/dart-sdk/bin/dart bin/test_runner.dart test -t run_release_test
../../bin/cache/dart-sdk/bin/dart bin/test_runner.dart test -t run_debug_test_android
```

Part of https://github.com/flutter/flutter/issues/117000

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/lib/tasks/run_tests.dart [hunk 1]
```diff
@@ -102,28 +102,34 @@ class AndroidRunOutputTest extends RunOutputTask {
 
   @override
   TaskResult verify(List<String> stdout, List<String> stderr) {
+    final String gradleTask = release ? 'assembleRelease' : 'assembleDebug';
+    final String apk = release ? 'app-release.apk' : 'app-debug.apk';
```
Comment (christopherfujino) [inline dev/devicelab/lib/tasks/run_tests.dart:106]: i'm assuming we never test in profile mode?

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
Can you remove `bringup: true` from the `run_debug_test_android` tests, let them run in presubmit to validate it passes? Then revert that before merging this?

**Comment by loic-sharma** [comment]
@jmagman Good idea! Tests passed here:

1. [Mac_android run_debug_test_android](https://ci.chromium.org/ui/p/flutter/builders/try/Mac_android%20run_debug_test_android/1/overview)
2. [Mac_arm64_android run_debug_test_android](https://ci.chromium.org/ui/p/flutter/builders/try/Mac_arm64_android%20run_debug_test_android/1/overview)

I'll revert back now to postsubmit bringup tests now.

File: dev/devicelab/lib/tasks/run_tests.dart (hunk 1 above)
Comment (loic-sharma) [reply dev/devicelab/lib/tasks/run_tests.dart:106]: Yup that's correct, but that might be worth adding in the future though

**Review by jmagman — APPROVED** [review]
LGTM!

**Pull request merged**